client/session.go

```diff
@@ -870,7 +867,7 @@ func (s *session) fetchAllAttempt(
 		resultErrs             int32
 		majority               int32
 		fetchBatchOpsByHostIdx [][]*fetchBatchOp
-		nsID                   = []byte(namespace)
+		nsID                   = namespace.Data().Get()
```

Comment:
Hm it's kind of unsafe to take a ref to the ts.ID underlying bytes and keep it around for a while.

In the `f.append(...)` call maybe we should make it take the `ts.ID` refs.  

Even then we can't guarantee the lifecycle of the ts.ID as this method might return early (say due to error or othe reason) but requests will be in flight with refs to the ts.IDs (both namespace and series IDs).

Maybe we should copy all the IDs and namespace ID at this point... either that or we need to use contexts to finalize of these properly and do the ref counting that way.